src/main/java/com/emc/object/s3/S3SignerV2.java

```diff
@@ -26,205 +26,54 @@
  */
 package com.emc.object.s3;
 
-import com.emc.object.s3.jersey.BucketFilter;
-import com.emc.object.s3.jersey.NamespaceFilter;
-import com.emc.object.s3.request.PresignedUrlRequest;
 import com.emc.object.util.RestUtil;
+import com.sun.jersey.api.client.ClientRequest;
 import org.apache.commons.codec.binary.Base64;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
-import javax.crypto.Mac;
-import javax.crypto.spec.SecretKeySpec;
-import java.io.UnsupportedEncodingException;
-import java.net.MalformedURLException;
-import java.net.URI;
-import java.net.URL;
-import java.security.InvalidKeyException;
-import java.security.NoSuchAlgorithmException;
+import java.nio.charset.StandardCharsets;
 import java.util.*;
 
-public final class S3SignerV2 {
-    private static final Logger log = LoggerFactory.getLogger(S3SignerV2.class);
-
-    private S3Config s3Config;
-    private SortedSet<String> signedParameters;
+public final class S3SignerV2 extends S3Signer {
 
     public S3SignerV2(S3Config s3Config) {
-        this.s3Config = s3Config;
-        signedParameters = new TreeSet<String>(Arrays.asList(
-                "acl", "torrent", "logging", "location", "policy", "requestPayment", "versioning",
-                "versions", "versionId", "notification", "uploadId", "uploads", "partNumber", "website",
-                "delete", "lifecycle", "tagging", "cors", "restore",
-                S3Constants.PARAM_RESPONSE_HEADER_CACHE_CONTROL,
-                S3Constants.PARAM_RESPONSE_HEADER_CONTENT_DISPOSITION,
-                S3Constants.PARAM_RESPONSE_HEADER_CONTENT_ENCODING,
-                S3Constants.PARAM_RESPONSE_HEADER_CONTENT_LANGUAGE,
-                S3Constants.PARAM_RESPONSE_HEADER_CONTENT_TYPE,
-                S3Constants.PARAM_RESPONSE_HEADER_EXPIRES,
-                S3Constants.PARAM_ENDPOINT,
-                S3Constants.PARAM_IS_STALE_ALLOWED));
-        if (s3Config.isSignMetadataSearch()) {
-            signedParameters.add(S3Constants.PARAM_QUERY);
-            signedParameters.add(S3Constants.PARAM_SEARCH_METADATA);
-        }
+        super(s3Config);
+    }
+
+    public void sign(ClientRequest request, String resource, Map<String, String> parameters, Map<String, List<Object>> headers) {
+        String stringToSign = getStringToSign(request.getMethod(), resource, parameters, headers);
+        String signature = getSignature(stringToSign, null);
+        RestUtil.putSingle(headers, "Authorization", "AWS " + s3Config.getIdentity() + ":" + signature);
     }
 
+    // this method is for testing
```

Comment:
what calls this method?  Can we refactor the test so it uses another method, so we don't have methods that only exists for testing?  seems like the test could use the other `sign()` method and construct an appropriate `ClientRequest` instance

Reply:
Changed.